compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -906,7 +917,40 @@ static void PrintGetServiceDescriptorMethod(const ServiceDescriptor* service,
     p->Indent();
     p->Print(*vars, "return $proto_class_name$.getDescriptor();\n");
     p->Outdent();
+    p->Print(*vars, "}\n\n");
+
+    p->Print(*vars, "@$Override$\n");
+    p->Print(
+        *vars,
+        "public com.google.protobuf.Descriptors.ServiceDescriptor getServiceDescriptor() {\n");
+    p->Indent();
+    p->Print(*vars, "return getFileDescriptor().findServiceByName(\"$service_name$\");\n");
+    p->Outdent();
+    p->Print(*vars, "}\n");
+
+    p->Outdent();
+    p->Print(*vars, "}\n\n");
+
+    p->Print(
+        *vars,
+        "private static final class $proto_method_descriptor_supplier$ extends $proto_file_descriptor_supplier$ implements $ProtoMethodDescriptorSupplier$ {\n");
```

Comment:
So far grpc generated code is fairly readable.  One of the things that makes it that way is generally fitting within line limits.  Until we auto format code, could this be wrapped to try to fit? Google style is binary ops go on the following line.  Something like:

```java
class Foo
    extends Bar
    implements Baz
```